lib/dao.js

```diff
@@ -433,6 +439,7 @@ DataAccessObject.create = function(data, options, cb) {
           if (err) return cb(err);
 
           if (connector.create.length === 4) {
+            options = addTransaction(Model, options);
             connector.create(modelName, obj.constructor._forDB(context.data), options, createCallback);
           } else {
             connector.create(modelName, obj.constructor._forDB(context.data), createCallback);
```

Comment:
When the connector does not support transactions, but the `options` or Model properties indicate that a transaction should be used, then I think we should throw an error. Otherwise our users will think their code is wrapped in a transaction while it is not. This applies to all DAO methods.

Reply:
My concern is that the connection between the fact whether connector supports transactions and the number of argument of CRUD methods like `connector.create` is not obvious and easy to unintentionally break in the future. I personally prefer a defensive approach - if we assume that transactions are enabled only when connectors support `options` argument, then it's better to capture/assert that assumption.

For example:

```js
options = addTransaction(Model, options);
if (connector.create.length === 4) {
   connector.create(modelName, ..., options, createCallback);
} else if (options.transaction) {
  throw new Error('The connector does not support create() within a transaction');
} else {
   connector.create(modelName, ..., createCallback);
}
```

Reply:
Ok, that makes sense. I could easily move that to `addTransaction()` and call the function `handleTransaction()` instead, so there wouldn't any replication of code. The name of the calling function (`create`) in this example) could be passed to it as a 2nd argument.

Reply:
I've just addressed this now in e3d47d8ebb1b77e77ea638e10a512dfadb0aa066 
Please have a look and let me know your thoughts.